guides/assets/esa-initial-flow.txt

```diff
@@ -0,0 +1,40 @@
+sequenceDiagram
+    participant User
+    participant App
+    participant Application Route
+    participant Authenticated Route
+    participant Unauthenticated Route
+    participant Session Service
+    participant Torii Provider
+    participant GitHub
+    participant Token Exchanger
+    User->>App: Use https://git-tool.com
+    App->>Application Route: transitionTo(application)
```

Comment:
> Should I just be more verbose in the actor names, e.g. Index Route (AuthenticatedRouteMixin)?

I think that sounds like a good idea.

> I think the points you're raising highlight that the diagram doesn't really stand on its own with the simplifications I made. It either requires embedding in an explanatory context or more elaboration in the diagram itself. What do you think?

I think we'll probably never be able to make the diagram stand on its own as it just includes so much information that it will always need some context I guess. I don't think that's a problem though as long as it works well in combination with the text? My main point was really only that a few things in there are slightly different from what actually happens in reality (e.g. there is no `transitionTo` when the user boots the app on the index route) and thus might be misleading.